### PR TITLE
Add missing comments in Vectors layer code

### PR DESCRIPTION
**Pull request by jules-vanaret**
# Description
This is a follow-up micro-PR to the discussion in #5740 about adding last comments to the Vectors layer code.

## Type of change
- [x] Enhancement (non-breaking improvements of an existing feature)
- [x] Documentation (update of docstrings and deprecation information)

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: napari/layers/vectors/vectors.py [hunk 1]
```diff
@@ -622,13 +622,23 @@ def edge_contrast_limits(
     @property
     def _view_face_color(self) -> np.ndarray:
         """(Mx4) np.ndarray : colors for the M in view vectors"""
+
+        # Create as many colors as there are visible vectors
         face_color = self.edge_color[self._view_indices]
         face_color[:, -1] *= self._view_alphas
```
Comment (brisvag) [inline napari/layers/vectors/vectors.py]: I'm noticing now: doesn't this `*=` actually change inplace the array in `color_manager`? I think this should be a copy maybe? @andy-sweet what do you think?

**Review by brisvag — APPROVED** [review]
LGTM. I noticed one possible issue (unrelated to this PR) so I pointed it out, but this can be merged regardless.

File: napari/layers/vectors/vectors.py (hunk 1 above)
Comment (andy-sweet) [reply napari/layers/vectors/vectors.py]: I think `Vectors._view_indices` is a list or `ndarray`, so the line here uses [fancy/advanced indexing](https://numpy.org/doc/stable/user/basics.indexing.html#advanced-indexing), which means we get a new array rather than a view of the underlying one in `ColorManager`.

That would actually be a very useful implementation comment to add here.

Comment (brisvag) [reply napari/layers/vectors/vectors.py]: Ah, that makes sense!

Comment (jules-vanaret) [reply napari/layers/vectors/vectors.py]: I added a few lines about that.

File: napari/layers/vectors/vectors.py [hunk 2]
```diff
@@ -621,14 +621,27 @@ def edge_contrast_limits(
 
     @property
     def _view_face_color(self) -> np.ndarray:
-        """(Mx4) np.ndarray : colors for the M in view vectors"""
+        """(Mx4) np.ndarray : colors for the M in view triangles"""
```
Comment (brisvag) [inline napari/layers/vectors/vectors.py:624]: Why this change? This is always the M vectors in view, even with the arrow vis, right? Otherwise it would be the "M*3 triangles representing the M vectors" in that case?

Comment (jules-vanaret) [reply napari/layers/vectors/vectors.py:624]: If I'm not mistaken this is supposed to give information about the output of `_view_face_color`, whose shape depends on the number of triangles drawn (actually, the final shape is `(N_triangles, 4)`.
If `M` is defined as the number of vectors, then yes I should write something like `(aMx4) np.ndarray: ... with a=2,1,3 for style='line'...`. 
But that's the point, I just saw that `M` was not defined precisely, so if you instead consider that `M` is the length of the output, then it is equal to the number of triangles, not vectors.

Comment (brisvag) [reply napari/layers/vectors/vectors.py:624]: Ok, I see... It's just a bit confusing that we have some stuff working on vector basis and some on triangle basis, my bad :)
